django/db/migrations/loader.py

```diff
@@ -186,15 +186,20 @@ def build_graph(self):
         # This is just for dependency re-pointing when applying replacements,
         # so we ignore run_before here.
         reverse_dependencies = {}
-        for key, migration in normal.items():
+        reverse_dependencies_normal = {}
+        for key, migration in self.disk_migrations.items():
             for parent in migration.dependencies:
                 reverse_dependencies.setdefault(parent, set()).add(key)
+                if key in normal:
+                    reverse_dependencies_normal.setdefault(parent, set()).add(key)
         # Remember the possible replacements to generate more meaningful error
         # messages
         reverse_replacements = {}
         for key, migration in replacing.items():
             for replaced in migration.replaces:
                 reverse_replacements.setdefault(replaced, set()).add(key)
+        # Set to store all replacing migrations that cannot be applied.
```

Comment:
Maybe you can enhance this comment with an example? I'm not sure if "inapplicable" is the best word choice (meaning "not relevant or appropriate" according to the first definition I found).